lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -2101,6 +2078,92 @@ class DecomposeAtenNumpyTOp : public OpRewritePattern<AtenNumpyTOp> {
 };
 } // namespace
 
+// Decompose aten.var(x, dims) into:
+// sub = aten.sub(x, aten.mean(x, dims))
+// square = aten.square(sub)
+// For Unbiased case:
+// out = aten.sum(square, dims) / (productDimSize-1)
+// For Biased case:
+// out = aten.mean(square, dims)
+namespace {
+class DecomposeAtenVarDimOp : public OpRewritePattern<AtenVarDimOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenVarDimOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.self();
+    Value dimList = op.dim();
+    Value keepDim = op.keepdim();
+    Type outputType = op.getType();
+    unsigned inputRank = getTensorRank(self);
+    BaseTensorType inputTensorTy = self.getType().cast<BaseTensorType>();
+    if (!inputTensorTy.hasDtype() ||
+        !inputTensorTy.getDtype().isa<mlir::FloatType>()) {
+      return rewriter.notifyMatchFailure(op,
+                                         "support floating type input only");
+    }
+
+    auto dimListConstruct = dimList.getDefiningOp<PrimListConstructOp>();
+    if (!dimListConstruct) {
+      return rewriter.notifyMatchFailure(
+          op, "expect dimList to be constructed from list construct");
+    }
+
+    // TODO: Add support for keepdim arg equal to `False` and dimList size not
+    // equal to input rank.
+    if (!keepDim.getType().isa<Torch::NoneType>()) {
+      bool keepdim;
+      if (!matchPattern(keepDim, m_TorchConstantBool(&keepdim)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: keepdim must be a constant");
+      else if (!keepdim && dimListConstruct.elements().size() != inputRank)
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: for keepdim = false, no. of reduction dims "
```

Comment:
What is the reason for this limitation? I would expect things to work since `AtenMeanDimOp` and `AtenSumDimIntListOp` take care of reducing the dimensions

Reply:
For the cases like, `Input size : [2,3,4], unbiased=False/True, dim=[0,2], keepdim=False`, https://github.com/llvm/torch-mlir/blob/645e7e6fe39a6c1a97de5d01ead2b9d324dbdffe/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L2133 will generate a tensor of size [3] only, which results in an error for https://github.com/llvm/torch-mlir/blob/645e7e6fe39a6c1a97de5d01ead2b9d324dbdffe/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L2135-L2136 because the `self tensor` is of size [2,3,4] and `meanAlongDims` is of size [3]. I think we don't have this kind of broadcasting support for the tensors in these ops lowering. That's why I have added this restriction.

Reply:
I think `keepDim` should always be `True` in `AtenMeanDimOp`, since at this point in the decomposition we are not yet reducing the rank of the input tensor.

Reply:
In that case, we have to manually create the `resultType` for `AtenMeanDimOp` when `keepDim=False`, since the result type of the Var op will not match the result type of the `AtenMeanDimOp`.

Reply:
Hi @ramiro050, I have removed this limitation from the code via some modifications. Thanks for your suggestions. Please review this PR so that it can be merged.